### PR TITLE
Decrypt all incoming omemo msg

### PR DESCRIPTION
**Pull request by paulfariello**

### CODE REVIEW

**Comment by paulfariello** [comment]
Addresses #1122

**Comment by jubalh** [comment]
For the record:
Currently messages are mostly passed as char*. We planned to have a ProfMessage struct to add more meta information that we sometimes need (like stanza Id, read state). This will make it easier to later implement last message correction or changing messages in the buffer. Also printing lines for last read or messages in different colours if they are unread.
This was planned for 0.8.0.

But since with this PR we already ran into something it was decided to create a first version of them now already. So we can store the enc_type of a message along with the message instead of passing as seperate parameter.

File: src/xmpp/message.h [hunk 1]
```diff
@@ -35,9 +35,34 @@
...
+   /* The raw body from xmpp message, either plaintext or OTR encrypted text */
+   char *body;
+   /* The encrypted message as for PGP */
+   char *encrypted;
+   /* The message that will be printed on screen and logs */
+   char *plain;
+   GDateTime *timestamp;
+   prof_enc_t enc;
+   gboolean trusted;
+} prof_message_t;
```
Comment (jubalh) [inline src/xmpp/message.h]: Actually I would do it like this:
typedef struct prof_message_t {
...
} ProfMessage;

I think all other structs also follow that convention. For example see prof_chat_win_t -> ProfChatWin.

File: src/xmpp/message.c [hunk 2]
```diff
@@ -999,38 +1040,39 @@ _handle_chat(xmpp_stanza_t *const stanza)
...
 
     // standard chat message, use jid without resource
-    xmpp_ctx_t *ctx = connection_get_ctx();
-    GDateTime *timestamp = stanza_get_delay(stanza);
-    if (!message && body) {
-        message = xmpp_stanza_get_text(body);
+    prof_message_t *message = message_init();
+    message->jid = jid;
+
+    message->timestamp = stanza_get_delay(stanza);
```
Comment (jubalh) [inline src/xmpp/message.c]: Will this always have a value or is it possible that sometimes we dont get one and need to get the receiving time ourselves? Of course this then could differ from when it was sent..

**Pull request merged**